yt/geometry/oct_visitors.pyx

```diff
@@ -338,3 +338,172 @@ cdef class MortonIndexOcts(OctVisitor):
                 np.uint64(coord[1]),
                 np.uint64(coord[2]))
         self.index += 1
+
+# Store cell index
+cdef class StoreIndex(OctVisitor):
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    cdef void visit(self, Oct* o, np.uint8_t selected):
+        if not selected: return
+        self.cell_inds[o.domain_ind, self.ind[2], self.ind[1], self.ind[0]] = self.index
+
+        self.index += 1
+
+cdef class BaseNeighbourVisitor(OctVisitor):
+    def __init__(self, OctreeContainer octree, int domain_id = -1):
+        self.octree = octree
+        self.neigh_ind = np.zeros(3, np.int8)
+        super(BaseNeighbourVisitor, self).__init__(octree, domain_id)
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    cdef void set_neighbour_info(self, Oct *o, int ishift[3]):
+        cdef int i
+        cdef np.float64_t c, dx
+        cdef np.int64_t ipos
+        cdef np.float64_t fcoords[3]
+        cdef Oct *neighbour
+        cdef bint local_oct
+        cdef bint other_oct
+        dx = 1.0 / ((1 << self.oref) << self.level)
+        local_oct = True
+
+        # Compute position of neighbouring cell
+        for i in range(3):
+            c = <np.float64_t> (self.pos[i] << self.oref)
+            fcoords[i] = (c + 0.5 + ishift[i]) * dx / self.octree.nn[i]
+            # Assuming periodicity
+            if fcoords[i] < 0:
+                fcoords[i] += 1
+            elif fcoords[i] > 1:
```

Comment:
can you make this explicitly a float?  and, shouldn't this be checking if it's greater than (and above, less than) the domain width?

Reply:
actually i may be wrong on that, but want confirmation.